### PR TITLE
Add `MessageCorrelation` record and intent

### LINKED ISSUE #20164: Create message correlation record and intents
<h2>Description</h2>

- Create a new `MessageCorrelationRecord` with properties:
  - `name`
  - `correlationKey`
  - `variables`
  - `tenantId`
- Create a new `MessageCorrelationIntent` with values:
  - `CORRELATE`
  - `CORRELATED`
  - `NOT_CORRELATED`

### LINKED ISSUE #20186: Create template for new message correlation record
<h2>Description</h2>

We msut create templates for the new record in the ES and OS exporter. Most of this can be copied from the [message record template](https://github.com/camunda/camunda/blob/main/zeebe/exporters/opensearch-exporter/src/main/resources/zeebe-record-message-template.json).

Differences are:
1. The name
2. No TTL
3. No Deadline
4. Add correlated boolean
5. Add process instance key

<h3>Dependencies</h3>

- #20164

### LINKED ISSUE #20189: Add new message correlation record to `CompactRecordLogger`
<h2>Description</h2>

The `CompactRecordLogger` is responsible for logging all our records. Since we've created a new record we'll have to make sure to add it here.

<h3>Dependencies</h3>

- #20164

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

This PR introduces the `MessageCorrelation` value type and intents. The record is similar to the `MessageRecord` except it has a few fields less. It shall be used for the purpose of a "synchronous" message publication.

## Related issues

closes #20164
closes #20189 
closes #20186

### CODE REVIEW

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/Engine.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public class Engine implements RecordProcessor {
       "Expected to process record '%s' without errors, but exception occurred with message '%s'.";
 
   private static final EnumSet<ValueType> SUPPORTED_VALUETYPES =
-      EnumSet.range(ValueType.JOB, ValueType.COMPENSATION_SUBSCRIPTION);
+      EnumSet.range(ValueType.JOB, ValueType.MESSAGE_CORRELATION);
```
Comment (nicpuppa) [inline zeebe/engine/src/main/java/io/camunda/zeebe/engine/Engine.java:49]: You have not forgotten anything  🤣

File: zeebe/test-util/src/main/java/io/camunda/zeebe/test/util/record/CompactRecordLogger.java [hunk 2]
```diff
@@ -830,6 +832,22 @@ private String summarizeCommandDistribution(final Record<?> record) {
...
+      result.append(" correlationKey: ").append(value.getCorrelationKey());
+    }
+
+    result.append(" processInstanceKey: ").append(value.getProcessInstanceKey());
+
+    result.append(summarizeVariables(value.getVariables()));
+
+    return result.toString();
+  }
+
```
Comment (nicpuppa) [inline zeebe/test-util/src/main/java/io/camunda/zeebe/test/util/record/CompactRecordLogger.java:850]: 🔧 This method can be refactored like that:

```suggestion
  private String summarizeMessageCorrelation(final Record<?> record) {
    final MessageCorrelationRecordValue value = (MessageCorrelationRecordValue) record.getValue();
    final String correlationKey = value.getCorrelationKey();

    StringBuilder result = new StringBuilder()
        .append("\"").append(value.getName()).append("\"");

    if (correlationKey != null && !correlationKey.isEmpty()) {
        result.append(" correlationKey: ").append(correlationKey);
    }

    result.append(" processInstanceKey: ").append(value.getProcessInstanceKey())
          .append(summarizeVariables(value.getVariables()));

    return result.toString();
}

```

**Review by nicpuppa — APPROVED** [review]
LGTM, I left only a minor comment. Please have a look 👀

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/Engine.java (hunk 1 above)
Comment (remcowesterhoud) [reply zeebe/engine/src/main/java/io/camunda/zeebe/engine/Engine.java:49]: The developer handbook saved me on this 😆

**Comment by remcowesterhoud** [comment]
This has some failing tests because of #20197. I'll merge that one into this one and then merge them to main together.

**Pull request merged**